ratis-test/src/test/java/org/apache/ratis/server/raftlog/segmented/TestSegmentedRaftLog.java

```diff
@@ -522,19 +522,20 @@ public void testSegmentedRaftLogStateMachineData() throws Exception {
 
       sm.blockFlushStateMachineData();
       raftLog.appendEntry(entries.get(next++));
-      {
-        sm.blockWriteStateMachineData();
-        final Thread t = startAppendEntryThread(raftLog, entries.get(next++));
-        TimeUnit.SECONDS.sleep(1);
-        Assert.assertTrue(t.isAlive());
-        sm.unblockWriteStateMachineData();
-        t.join();
-      }
+
+      sm.blockWriteStateMachineData();
+      final Thread t = startAppendEntryThread(raftLog, entries.get(next++));
+      TimeUnit.SECONDS.sleep(1);
+      Assert.assertTrue(t.isAlive());
+      sm.unblockWriteStateMachineData();
+
       assertIndices(raftLog, flush, next);
       TimeUnit.SECONDS.sleep(1);
       assertIndices(raftLog, flush, next);
       sm.unblockFlushStateMachineData();
       assertIndicesMultipleAttempts(raftLog, flush + 2, next);
+
+      t.join();
```

Comment:
Why do we need this change?

Reply:
It is because `raftLog.appendEntry(entry).get();` won't return until `sm.unblockFlushStateMachineData();` called.

Reply:
Got it. I think the old changes were good. Can you please bring it back?